components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { PropsWithChildren, useCallback, useEffect, useMemo, useRef, useState } from 'react';
+
+import css from './Dropdown.module.scss';
+
+export enum Placement {
+  Bottom = 'bottom',
+  BottomLeft = 'bottomLeft',
+  BottomRight = 'bottomRight',
+  Left = 'left',
+  LeftBottom = 'leftBottom',
+  LeftTop = 'leftTop',
+  Right = 'right',
+  RightTop = 'rightTop',
+  RightBottom = 'rightBottom',
+  Top = 'top',
+  TopLeft = 'topLeft',
+  TopRight = 'topRight',
+}
+
+interface Props {
+  content: React.ReactNode;
+  disableAutoDismiss?: boolean;
+  /** to hide dropdown with parent component state */
+  hide?: boolean;
+  initVisible?: boolean;
+  offset?: { x: number, y: number };
+  /** to set hide prop from parent component state to false */
+  onShow?: () => void;
```

Comment:
can we use onVisibleChange for this?

nit: `visible` seems more commonly used in our codebase.